api/scpca_portal/loader.py

```diff
@@ -0,0 +1,245 @@
+    submitter_whitelist: Set[str],
+    input_bucket_name: str,
+    reload_existing: bool,
+    update_s3: bool,
+) -> Project | None:
+    """
+    Validate that a project can be processed, creates it, and return the newly created project.
+    """
+    if not _can_process_project(project_metadata, submitter_whitelist):
+        return
+
+    # If project exists and cannot be purged, then throw a warning
+    project_id = project_metadata["scpca_project_id"]
+    if project := Project.objects.filter(scpca_id=project_id).first():
+        # If there's a problem purging an existing project, then don't process it
+        if _can_purge_project(project, reload_existing=reload_existing):
+            # Purge existing projects so they can be re-added.
+            logger.info(f"Purging '{project}")
+            project.purge(delete_from_s3=update_s3)
+        else:
+            return
+
+    logger.info(f"Importing Project {project_metadata['scpca_project_id']} data")
+    project = Project.get_from_dict(project_metadata)
+    project.s3_input_bucket = input_bucket_name
+    project.save()
+
+    return project
+
+
+def bulk_create_project_relations(project_metadata: Dict[str, Any], project: Project) -> None:
+    """
+    Create Contact, ExternalAccession and Publication instances from the project metadata dict,
+    and create a foreign key association with their project.
+    """
+    Contact.bulk_create_from_project_data(project_metadata, project)
+    ExternalAccession.bulk_create_from_project_data(project_metadata, project)
+    Publication.bulk_create_from_project_data(project_metadata, project)
+
+
+def create_samples_and_libraries(project: Project) -> None:
+    """Create all samples and libraries associated with the passed project."""
+    project.load_metadata()
+    if samples_count := project.samples.count():
+        logger.info(f"Created {samples_count} sample{pluralize(samples_count)} for '{project}'")
+
+
+def _prep_project_for_computed_file_generation(
+    project: Project, update_s3: bool = False
+) -> Project:
+    """
+    Purge all of a project's computed files (and optionally delete them from s3),
+    before new ones are generated.
+    """
+    project.purge_computed_files(update_s3)
+    return project
+
+
+def get_project_for_computed_file_generation(
+    project_id: str = "", update_s3: bool = False
+) -> Project:
+    """Retrieve, prep and return project for computed file generation."""
+    return _prep_project_for_computed_file_generation(
+        Project.objects.filter(scpca_id=project_id), update_s3
+    )
+
+
+def get_projects_for_computed_file_generation(update_s3: bool = False) -> List[Project]:
+    """
+    Query the database, prep and return all projects that need computed files to be generated.
+    """
+    return [
+        _prep_project_for_computed_file_generation(project, update_s3)
+        for project in Project.objects.filter(project_computed_files__isnull=True)
+    ]
```

Comment:
```suggestion
```

Let's remove these and instead query the project in the commands (handle missing project there). Then purging should be handled per script.